pkg/alignchecker/alignchecker.go

```diff
@@ -143,3 +145,86 @@ func check(name string, toCheck []reflect.Type, structs map[string]structInfo, c
 
 	return nil
 }
+
+// Read ELF symbols from .rodata and compare the values against given toCheck values
+func valueCheck(pathToObj string, toCheck map[string]reflect.Value) error {
+	f, err := elf.Open(pathToObj)
+	if err != nil {
+		return fmt.Errorf("ELF failed to open %s: %s", pathToObj, err)
+	}
+	defer f.Close()
+
+	elf_syms, err := f.Symbols()
+	if err != nil {
+		return fmt.Errorf("Failed to get ELF symbols")
+	}
+
+	rodata := f.Section(".rodata")
+	if rodata == nil {
+		return fmt.Errorf("Failed to open .rodata section")
```

Comment:
In Golang, we generally create errors that start with a lowercase. Also, if we are not formatting a message, then we want to just state what the error actually is (in form of a sentinel error variable). 

In this specific case, we indeed failed to open the `.rodata` section, but I presume it's because it's missing (?). So creating a sentinel error variable for this would be more ideal.

```
... (in global scope)
var ErrMissingROData = errors.New("missing .rodata")
...

rodata := f.Section(".rodata")
if rodata == nil {
	return ErrMissingROData
}
...
```

Reply:
Thanks. I went a slightly different way, but I think it's in the same spirit.